From 57aa966c69d3e447bc4924d0c85a0a4fd0cf2483 Mon Sep 17 00:00:00 2001
From: Tuyen Dang <tuyen.dang.xa@renesas.com>
Date: Mon, 26 May 2025 21:08:43 +0700
Subject: [PATCH 48/52] arm64: dts: renesas: r8a78000: Add SD/MMC node

Add a device node for SD/MMC on Renesas R-Car X5H (R8A78000) SoC.

Signed-off-by: Tuyen Dang <tuyen.dang.xa@renesas.com>
Signed-off-by: Khanh Le <khanh.le.xr@renesas.com>
Signed-off-by: Hai Pham <hai.pham.ud@renesas.com>
---
 arch/arm/dts/r8a78000.dtsi | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/arch/arm/dts/r8a78000.dtsi b/arch/arm/dts/r8a78000.dtsi
index c89faa25a6a..165a41b8504 100644
--- a/arch/arm/dts/r8a78000.dtsi
+++ b/arch/arm/dts/r8a78000.dtsi
@@ -95,6 +95,14 @@
 		clock-frequency = <0>;
 	};
 
+	mmc_ref_clk: mmc_ref_clk {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-frequency = <52000000>;
+		clock-output-names = "mmc";
+	};
+
+
 	soc: soc {
 		compatible = "simple-bus";
 		interrupt-parent = <&gic>;
@@ -573,6 +581,15 @@
 			#power-domain-cells = <0>;
 			#reset-cells = <1>;
 		};
+
+		mmc0: mmc@c0880000 {
+			compatible = "renesas,rcar-gen5-sdhi";
+			reg = <0 0xc0880000 0 0x2000>;
+			interrupts = <GIC_SPI 894 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&mmc_ref_clk>;
+			max-frequency = <200000000>;
+			status = "disabled";
+		};
 	};
 
 	timer {
-- 
2.34.1

